src/coreclr/scripts/superpmi_asmdiffs.py

```diff
@@ -170,8 +170,10 @@ def main(main_args):
     arch_name = coreclr_args.arch
     host_arch_name = "x64" if arch_name.endswith("64") else "x86"
     os_name = "universal" if arch_name.startswith("arm") else os_name
-    base_jit_path = os.path.join(coreclr_args.base_jit_directory, 'clrjit_{}_{}_{}.dll'.format(os_name, arch_name, host_arch_name))
-    diff_jit_path = os.path.join(coreclr_args.diff_jit_directory, 'clrjit_{}_{}_{}.dll'.format(os_name, arch_name, host_arch_name))
+    base_checked_jit_path = os.path.join(coreclr_args.base_jit_directory, "checked", 'clrjit_{}_{}_{}.dll'.format(os_name, arch_name, host_arch_name))
```

Comment:
I wouldn't do the tp-diff as part of `superpmi_asmdiffs.py` but would write a separate script exclusively for tp-diff. That way you can have a separate CI pipeline flow just to tpdiff without penalizing the asmdiff execution time.

I wouldn't even recommend having a parameter like `asmdiff` vs. `tpdiff` to this file, because it just pollutes the logic and hard to refactor in future.

Reply:
I see no reason to introduce yet another pipeline if we want to run TP impact on every PR. It will require even more resources (e.g. redownloading all collections again), it will force JIT devs to look in even more places to evaluate their change, and it does not scale (e.g. I want to add memory stats measurements in the future as well).

I really hope instead to rename "asmdiffs" to "diffs" (in fact, I have done this in a couple of places) and do all the base-diff measurements in this pipeline.

Reply:
FWIW, these pipelines spend way more time checking out sources, copying files and building than they do inside superpmi_asmdiffs.py. Currently (before this change) the x64 build spends only around 25 minutes in helix, so only around 1/3 of the build is actually spent inside superpmi_asmdiffs.py. So I do not think it is a bad idea to let it do more work and amortize some of all the other cost.

Reply:
>  it will force JIT devs to look in even more places to evaluate their change

+1 on that. Yeah, that justification sounds reasonable to have a single pipeline with both asmdiff/TPdiff. Also, while you are there, and if you have time, I would really want to surface any SPMI replay failures to extension page so we don't have to scan through the log.

Reply:
Let me look into it separately from this PR. I think we should report all of this back from helix/superpmi as JSON so that the summarize script can do a much better job of laying things out, but it will probably take a bit of work.